mteb/abstasks/AbsTask.py

```diff
@@ -126,43 +127,23 @@ def filter_languages(
             script: list of scripts to filter the task by. Will be ignored if language code specified the script. If None, all scripts are included.
                 If the language code does not specify the script the intersection of the language and script will be used.
         """
-        lang_script_codes = set()
-        # normalize to 3 letter language codes
-        normalized_langs = set()
-        filter_lang = languages is not None
-
-        if filter_lang:
-            for lang in languages:
-                lang_script = lang.split("-")
-
-                is_lang_script_code = len(lang_script) == 2
-                if is_lang_script_code:
-                    normalized_langs.add(lang_script[0])
-                    lang_script_codes.add(lang)
-                else:
-                    normalized_langs.add(lang)
-
-        filter_scripts = script is not None
-        script_codes: set[str] = set(script) if filter_scripts else set()
+        lang_script_filter = LangScriptFilter.from_languages_and_scripts(
```

Comment:
I'm not entirely sure I like this from a locality of behaviour standpoint. Up until now I could just look at the code her and understand what it did, now I have to look in another file in a dataclass for behaviour that directly affects what is happening here.

I might misunderstand the intention here, but I think this method would pretty much do the same thing and is (at least to my eyes) a bit easier to understand.
```python
def filter_languages(
    self, languages: list[str] | None, script: list[str] | None = None
) -> AbsTask:
    splits_to_keep: list[str] = []
    if not isinstance(self.metadata.eval_langs, dict):
        self.langs = self.metadata.eval_langs
        return self
    for hf_subset, langs in self.metadata.eval_langs.items():
        for langscript in langs:
            _lang, _script = langscript.split("-")
            if (languages is None) or (_lang in languages) or (langscript in languages):
                if (script is None) or (_script in script):
                    splits_to_keep.append(hf_subset)
    self.langs = splits_to_keep
    return self
```

Reply:
> I'm not entirely sure I like this from a locality of behaviour standpoint. Up until now I could just look at the code her and understand what it did, now I have to look in another file in a dataclass for behaviour that directly affects what is happening here.

The current format specified `langscripts.contains_language(language) or langscripts.contains_language(script)`. I believe that is more clear than:
`(languages is None) or (_lang in languages) or (langscript in languages)`

As mentioned previously the main idea was also to re-use the functionality. It also allows me to test the functionality directly in a test.

I am also not entirely sure it is equivalent. I would have to test it. E.g. I believe script = ["Cyrl"], language = ["eng-Latn"] with input "eng-Latn" would fail even though it shouldn't.

Reply:
Yeah I wasn't sure about what it was doing exactly, like what cases are included and what cases aren't so my code might not do that correctly.
Regarding the point about clarity: I'll try not to be radical, I think the solution you propose is fine, it's just that I don't necessarily like that I can't know what `contains_language` does until I look.

Reply:
I think it is a totally fair point - Style is important but hard to get right good with some discussion